components/camel-sip/src/main/java/org/apache/camel/component/sip/listener/SipPresenceAgentListener.java

```diff
@@ -179,6 +187,64 @@ public void processSubscribe(RequestEvent requestEvent,
         }
     }
 
+    public void processMessage(RequestEvent requestEvent, ServerTransaction serverTransactionId) {
+        Request request = requestEvent.getRequest();
+        try
+        {
+            LOG.debug("SipPresenceAgentListener: Received an MESSAGE request");
+            LOG.debug("SipPresenceAgentListener retrieved MESSAGE: {}", requestEvent.getRequest());
+
+            Response response = null;
+            MessageFactory messageFactory = sipPresenceAgent.getConfiguration().getMessageFactory();
+            boolean contentTypeSupported = false;
+
+            SipProvider sipProvider = (SipProvider) requestEvent.getSource();
+            String contentType = request.getHeader("Content-Type").toString();
+
+            if(contentType == null)
+            {
+                response = messageFactory.createResponse(400, request); // bad request
+                sipProvider.sendResponse(response);
+                return;
+            }
+            else if(contentType.contains("text/plain"))
```

Comment:
@nikvaessen could you allow other content types too?  They can be accepted by Camel as binary data (byte[]), just make sure you store the content type name in the header ```Exchange.CONTENT_TYPE```